tensorflow_quantum/python/optimizers/rotosolve_minimizer.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2020 The TensorFlow Quantum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""The rotosolve minimization algorithm
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import math
+import collections
+import tensorflow as tf
+
+
+def prefer_static_shape(x):
+    """Return static shape of tensor `x` if available,
+     else `tf.shape(x)`.
+    Args:
+      x: `Tensor` (already converted).
+    Returns:
+      Numpy array (if static shape is obtainable), else `Tensor`.
+    """
+    return prefer_static_value(tf.shape(x))
+
+
+def prefer_static_value(x):
+    """Return static value of tensor `x` if available, else `x`.
+    Args:
+      x: `Tensor` (already converted).
+    Returns:
+      Numpy array (if static value is obtainable), else `Tensor`.
+    """

```

Comment:
ditto.

Reply:
Fixed in 112400e847d6d98588c1708517aed5b7b7aeeb43